crates/configuration/src/relaychain.rs

```diff
@@ -0,0 +1,118 @@
+use crate::shared::{
+    node::NodeConfig,
+    types::{Arg, Command, ContainerImage, DbSnapshot, Resources},
+};
+
+/// A relaychain configuration, composed of nodes and fine-grained configuration options.
+pub struct RelaychainConfig {
+    /// Default command to run the node. Can be overriden on each node.
+    default_command: Option<Command>,
+
+    /// Default image to use (only podman/k8s). Can be overriden on each node.
+    default_image: Option<ContainerImage>,
+
+    /// Default resources. Can be overriden on each node.
+    default_resources: Option<Resources>,
+
+    /// Default database snapshot. Can be overriden on each node.
+    default_db_snapshot: Option<DbSnapshot>,
+
+    /// Chain to use (e.g. rococo-local).
+    chain: String,
+
+    /// Chain specification JSON file to use.
+    chain_spec_path: Option<String>,
+
+    /// Default arguments to use in nodes. Can be overriden on each node.
+    default_args: Vec<Arg>,
+
+    /// Set the count of nominators to generator (used with PoS networks).
+    random_nominators_count: Option<u32>,
+
+    /// Set the max nominators value (used with PoS networks).
+    max_nominations: Option<u16>,
+
+    /// Nodes to run.
+    nodes: Vec<NodeConfig>,
+    // [TODO]: do we need node_groups in the sdk?
+    // node_groups?: NodeGroupConfig[];
+
+    // [TODO]: allow customize genesis
+    // genesis?: JSON | ObjectJSON;
+}
+
+impl Default for RelaychainConfig {
+    fn default() -> Self {
+        // [TODO]: define the default value for a relaychain
+        todo!()
+    }
+}
+
+impl RelaychainConfig {
+    pub fn with_default_command(self, command: Command) -> Self {
+        Self {
+            default_command: Some(command),
+            ..self
+        }
+    }
+
+    pub fn with_default_image(self, image: ContainerImage) -> Self {
+        Self {
+            default_image: Some(image),
+            ..self
+        }
+    }
+
+    pub fn with_default_resources(self, f: fn(Resources) -> Resources) -> Self {
+        Self {
+            default_resources: Some(f(Resources::default())),
+            ..self
+        }
+    }
+
+    pub fn with_default_db_snapshot(self, db_snapshot: DbSnapshot) -> Self {
+        Self {
+            default_db_snapshot: Some(db_snapshot),
+            ..self
+        }
+    }
```

Comment:
I do not think that there are default values; Meaning if one do not supply then there is no "default replacement"

Reply:
There is an implementation of the ```Default``` trait to be determined so we can know what we expect as good defaults:

```rust
impl Default for RelaychainConfig {
    fn default() -> Self {
        // [TODO]: define the default value for a relaychain
        todo!()
    }
}
```

Reply:
I mean - there cannot be a default image, when not provided. Or a default command. User always needs to provide those. the only default value I remember is the `cumulus_based` which defaults to `true`